compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -238,38 +238,40 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
         !owner.isEmptyPackage || ctx.owner.enclosingPackageClass.isEmptyPackage
       }
 
+    import BindingPrec.*
+    type Result = (Type, BindingPrec)
```

Comment:
I don't think we should use a boxed type here. `findRefRecur` is very hot, we should avoid allocating where we can.